dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ContainerClassLoader.java

```diff
@@ -1663,28 +1666,31 @@ protected void addLibraryFile(File f) {
         //this area subject to refactor following shared lib rework..
         //ideally the shared lib code will start passing us ArtifactContainers, and it
         //will own the management of the ACF via DS.
-
-        //NASTY.. need to use DS to get the ACF, not OSGi backdoor ;p
         BundleContext bc = FrameworkUtil.getBundle(ContainerClassLoader.class).getBundleContext();
-        ServiceReference<ArtifactContainerFactory> acfsr = bc.getServiceReference(ArtifactContainerFactory.class);
-        if (acfsr != null) {
-            ArtifactContainerFactory acf = bc.getService(acfsr);
-            if (acf != null) {
-                //NASTY.. using this bundle as the cache dir location for the data file..
-                try {
-                    ArtifactContainer ac = acf.getContainer(bc.getBundle().getDataFile(""), f);
+        File dataFile = null;
+        try {
+            dataFile = bc == null ? null : bc.getDataFile("");
+        } catch (IllegalStateException e) {
+            Tr.debug(tc, "Invalid context. Liberty is likely shutting down.");
```

Comment:
Did you intend not to include the exception in the debug message?
`Tr.debug(tc, "Invalid context. Liberty is likely shutting down.", e);`

Reply:
I admittedly did not consider this very much because the likelihood of it happening is pretty slim.  The BundleContext would have to become invalid from the time we got to the point we call `getDataFile`.  Not sure it is worth it to add the exception there.

Reply:
Good enough.  FWIW I've never seen the exception in the field.